### PR TITLE
refactor: skip idling when timeout is 0

### PR DESCRIPTION
**Pull request by xcoulon**
no need to report an error, since timeout=0 is valid for '___noidling' tiers

see also:
- https://github.com/codeready-toolchain/api/pull/349
- discussion on [Slack#forum-dev-sandbox](https://redhat-internal.slack.com/archives/CHK0J6HT6/p1677582613636039)

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: controllers/idler/idler_controller_test.go [hunk 1]
```diff
@@ -343,28 +343,47 @@ func TestEnsureIdling(t *testing.T) {
...
-			require.NoError(t, err)
-			assert.Equal(t, reconcile.Result{}, res)
-			memberoperatortest.AssertThatIdler(t, idler.Name, cl).HasConditions(memberoperatortest.FailedToIdle("timeoutSeconds should be bigger than 0"))
+		// when
+		res, err := reconciler.Reconcile(context.TODO(), req)
+
+		// then
+		require.NoError(t, err)
+		assert.Equal(t, reconcile.Result{}, res)
+		memberoperatortest.AssertThatIdler(t, idler.Name, cl).HasNoConditions()
```
Comment (mfrancisc) [inline controllers/idler/idler_controller_test.go:365]: I was wondering if setting a condition like `NoIdling` in this case could make sense, if it makes it more clear when debugging/checking the objects in the cluster 🤷‍♂️

Comment (rajivnathan) [reply controllers/idler/idler_controller_test.go:365]: +1

**Review by alexeykazakov — APPROVED** [review]

File: controllers/idler/idler_controller_test.go (hunk 1 above)
Comment (xcoulon) [reply controllers/idler/idler_controller_test.go]: done in 33b2b1c

Comment (mfrancisc) [reply controllers/idler/idler_controller_test.go]: @xcoulon thanks for addressing it! 

And good catch regarding the mixed usage of `v1` and `metav1` package !

**Pull request merged**